### DISCUSSION TITLE
Issue with Distributed training on 8 GPUs, empty logs

### DISCUSSION THREAD

**Discussion by scape7575** [coqui-ai/TTS#2646] [General Q&A]
So I have been trying to get distributed training working with Tacotron2, GlowTTS, VITS, and FastSpeech2, all to the same end. What happens is that It starts training as normal, but upon checking the trainer_1_log.txt there is nothing inside, while trainer_0_log.txt has logs. I am unsure if the this means that it is only training on one GPU or more, as all the other logs other than trainer_0_log.txt are empty.

> Training Environment:
 | > Current device: 0
 | > Num. of GPUs: 2
 | > Num. of CPUs: 80
 | > Num. of Torch Threads: 40
 | > Torch seed: 54321
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 > Start Tensorboard: tensorboard --logdir=/root/recipes/ljspeech/glow_tts/run-May-24-2023_12+29PM-0000000

**Upon using the nvidia-smi command, I can see that there is ram being taken by the training from all the gpus mentioned from the command**

Can anyone help me with fixing this? Or perhaps knows whats causing this issue or have run into something similar? I have been trying to get it to work for a month unfortunately

**Comment by erogol** [comment]
You can check nvidia-smi to see all GPUs run or not

**Reply by scape7575** [reply]
Hi, thank you for the reply. I have checked the nvidia-smi, all GPUs are running and the training script is taking VRAM from all GPUs, but what has gotten me confused is the lack of logs on all other trainer logs except trainer 0

**Reply by erogol** [reply]
there might be an issue with the trainer. but if they are all active don't worry about the logs.

**Reply by scape7575** [reply]
So it might be running fine, but there is just an issue with the logs being produced? Also I have noticed that the training speed with 8xV100 GPUs is the same as 1xV100 GPU, again unsure if the same issue is causing this. Thanks!
